docs/5.0/application-access.md

```diff
@@ -0,0 +1,264 @@
+  nodename: wellington
+  # The Auth token `tctl tokens add --type=app` or the static token. 4c7e15
+  auth_token: "4c7e15"
+  # This is the location of the Teleport Auth Server or Public Proxy
+  auth_servers:
+    - teleport.example.com:3080
+auth_service:
+  enabled: no
+proxy_service:
+  enabled: no
+ssh_service:
+   enabled: no
+# The app_service is new
+app_service:
+   enabled: yes
+    debug_app: true
+   apps:
+   - name: "internal-app"
+     uri: "http://10.0.1.27:8000"
+   - name: "kubernetes-dashboard"
+     # This version requires a public_addr for all Apps, these
+     #  applications should have a certificate and DNS setup
+     public_addr: "example.com"
+     # Optional Labels
+     labels:
+        name: "jwt"
+     # Optional Dynamic Labels
+     commands:
+     - name: "os"
+       command: ["/usr/bin/uname"]
+       period: "5s"
+   - name: "arris"
+     uri: "http://localhost:3001"
+     public_addr: "arris.example.com"
+   # Teleport Application Access can be used to proxy any HTTP Endpoint
+   # Note: Name can't include any spaces
+   - name: "hackernews"
+     uri: "https://news.ycombinator.com"
+     public_addr: "hn.example.com"
+```
+
+## Advanced Options
+
+### Customize Public Address
+
+```yaml
+   - name: "jira"
+     uri: "https://localhost:8001"
+     public_addr: "jira.example.com"
+```
+
+### Deeplink to Subdirectory
+Some Applications are available on a Subdirectory, examples include Kubernetes Dashboard.
+
+```yaml
+   - name: "k8s"
+     uri: "http://10.0.1.60:8001/api/v1/namespaces/kube-system/services/https:kubernetes-dashboard:/proxy/#/overview?namespace=default"
+     public_addr: "k8s.example.com"
+```
+### Rewrite
+We provide simple rewrites. This is helpful for applications
+
+```yaml
+   - name: "jira"
+     uri: "https://localhost:8001"
+     public_addr: "jira.example.com"
+     rewrite:
+        # Rewrite the "Location" header on redirect responses replacing the
+        # host with the public address of this application.
+        redirect:
+           - "localhost"
+           - "jenkins.internal.dev"
+```
+
+## View Applications in Teleport
+
+`https://[cluster-url]:3080/web/cluster/[cluster-name]/apps`
+
+
+## Integrating with JWTs
+
+### Introduction to JWTs
+JSON Web Token (JWT) is an open standard that defines a secure way to transfer information bewteen parties as a JSON Object.
+
+For a in-depth explanation please visit [https://jwt.io/introduction/](https://jwt.io/introduction/)
+
+Teleports JSON Web Token (JWT) includes three sections:
+
++ Header
++ Payload
++ Signature
+
+### Header
+
+**Example Header**
+```json
+{
+  "alg": "RS256",
+  "typ": "JWT"
+}
+```
+
+### Payload
+
+**Example Payload**
+
+```json
+{
+  "aud": [
+    "http://127.0.0.1:34679"
+  ],
+  "exp": 1603943800,
+  "iss": "aws",
+  "nbf": 1603835795,
+  "roles": [
+    "admin"
+  ],
+  "sub": "benarent",
+  "username": "benarent"
+}
+```
+**TODO Add info about claims etc.
+
+Teleport will
```

Comment:
Customers are asking about token TTL and validation. What happens when token expires @russjones

Reply:
Token TTL matches the certificate TTL exactly and validation happens with the keys from the jwks endpoint.

Internal applications should never receive a request with a expired token, because by then the certificate will have been revoked and the request will not be forwarded to internal application